netx/httptransport/http3transport_test.go

```diff
@@ -20,6 +21,97 @@ func (d MockHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *qui
 	return quic.DialAddrEarly(host, tlsCfg, cfg)
 }
 
+type MockSNIHTTP3Dialer struct {
+	namech chan string
+}
+
+func (d MockSNIHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+	d.namech <- tlsCfg.ServerName
+	close(d.namech)
```

Comment:
```suggestion
```

Because of the way in which you're using channels, it seems to me you don't need to close the channel here.